utils/build-script-impl

```diff
@@ -2294,14 +2309,12 @@ for host in "${ALL_HOSTS[@]}"; do
                     continue
                 fi
                 echo "--- Running tests for ${product} ---"
-                set -x
                 build_dir=$(build_directory ${host} ${product})
                 XCTEST_BUILD_DIR=$(build_directory ${host} xctest)
-                pushd "${FOUNDATION_SOURCE_DIR}"
-                ${NINJA_BIN} TestFoundation
-                LD_LIBRARY_PATH="$(get_host_install_destdir ${host})$(get_host_install_prefix ${host})"/lib/swift/:"${build_dir}/Foundation":"${XCTEST_BUILD_DIR}":${LD_LIBRARY_PATH} "${build_dir}"/TestFoundation/TestFoundation
-                popd
-                { set +x; } 2>/dev/null
+                call pushd "${FOUNDATION_SOURCE_DIR}"
+                call ${NINJA_BIN} TestFoundation
+                call env LD_LIBRARY_PATH="$(get_host_install_destdir ${host})$(get_host_install_prefix ${host})"/lib/swift/:"${build_dir}/Foundation":"${XCTEST_BUILD_DIR}":${LD_LIBRARY_PATH} "${build_dir}"/TestFoundation/TestFoundation
```

Comment:
Absolutely -- as long as it works.  Feel free to do a conservative `with_pushd` for now.